common/presenters/move-to-message-banner-component.js

```diff
@@ -0,0 +1,147 @@
+const { isEmpty, kebabCase } = require('lodash')
+
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const frameworkToTaskListComponent = require('./framework-to-task-list-component')
+
+function _returnComponent(title, content) {
+  return {
+    allowDismiss: false,
+    classes: 'app-message--instruction govuk-!-padding-right-0',
+    title: {
+      text: title,
+    },
+    content: {
+      html: content,
+    },
+  }
+}
+
+function moveToMessageBannerComponent({
```

Comment:
I think this is the main point of discussion. I've tried to extract the logic that was being done in the template to a presenter so that it simply passes the relevant component structure to the view. This means it can also be tested faster using unit tests.

I've tried to use a return early pattern but it does still feel quite complex to cater for the different scenarios.

One thing I was considering was extracting the contents of each if block out to separate methods/presenters but I wasn't sure if that would make it harder to follow.

Any thoughts around how to structure this method would be very welcome :)